test/E2ETests/TestAssets/DataRowTestProject/BaseClass.cs

```diff
@@ -16,5 +16,26 @@ public virtual void DataRowTestMethod(string a)
         {
             Assert.IsTrue(true);
         }
+
+        [TestMethod]
+        [DataRow(42)]
+        [DataRow(42, "BaseOptionalString1")]
+        [DataRow(42, "BaseOptionalString2", "BaseOptionalString3")]
+        [DataRow(42, "BaseOptionalString4", "BaseOptionalString5")]
+        public virtual void DataRowTestMethodWithSomeOptionalParameters(int i, string s1 = null, string s2 = null)
```

Comment:
Please add negative test cases as well, something where user specifies less than required arguments or more than required arguments and as a result of which the test method fails with appropriate exception.

Reply:
added with DataRowOptionalInvalidArguments